lib/gui/app.js

```diff
@@ -85,22 +86,41 @@ app.run(() => {
 app.run((AnalyticsService, ErrorService, UpdateNotifierService, SelectionStateModel) => {
   AnalyticsService.logEvent('Application start');
 
-  if (UpdateNotifierService.shouldCheckForUpdates() && !process.env.ETCHER_DISABLE_UPDATES) {
-    AnalyticsService.logEvent('Checking for updates');
+  const shouldCheckForUpdates = UpdateNotifierService.shouldCheckForUpdates();
+
+  if (shouldCheckForUpdates && !process.env.ETCHER_DISABLE_UPDATES) {
+
+    AnalyticsService.logEvent('Checking for updates', {
+      currentVersion: packageJSON.version
+    });
 
     UpdateNotifierService.isLatestVersion().then((isLatestVersion) => {
 
+      if (isLatestVersion) {
+        AnalyticsService.logEvent('Running latest version');
+        return;
+      }
+
       // In case the internet connection is not good and checking the
       // latest published version takes too long, only show notify
       // the user about the new version if he didn't start the flash
       // process (e.g: selected an image), otherwise such interruption
       // might be annoying.
-      if (!isLatestVersion && !SelectionStateModel.hasImage()) {
-
-        AnalyticsService.logEvent('Notifying update');
+      if (!SelectionStateModel.hasImage()) {
```

Comment:
Can we remove the negation from this condition, putting the event below here, and the actual `.notify()` call below?